pkgs/native_assets_cli/lib/src/link_mode.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'api/asset.dart';
+import 'api/build_config.dart';
+
+/// The link mode for a [NativeCodeAsset].
+///
+/// Known linking modes:
+///
+/// * [StaticLinking]
+/// * [DynamicLoading]
+///   * [LookupInProcess]
+///   * [LookupInExecutable]
+///   * [DynamicLoadingBundled]
+///   * [DynamicLoadingSystem]
+///
+/// See the documentation on the above classes.
+sealed class LinkMode {
```

Comment:
Using `sealed` means it would be a breaking change to add a link mode.
Also, it requires manual rolls into the Dart/Flutter SDKs, preventing us to roll and support a new link mode in a separate PR.

It's not marked `abstract` so we can construct this class now? That seems wrong, only subcassles should be constructible.

Use `abstract final` instead.

Reply:
Done.